src/schema/user/user.resolver.ts

```diff
@@ -72,6 +72,40 @@ export default class UserResolvers {
     }
   }
 
+  /*
+    getUserByPhone query takes a phoneNumber as a parameter and returns the
+    corresponding user
+  */
+  @Query(() => User, {
+    nullable: true,
+    description:
+      'Takes an array of User ObjectIDs as a parameter and returns an array of corresponding users. If an empty array is passed, All the users are returned.',
+  })
+  async getUserByPhone(
+    @Arg('phoneNo')
+    phoneNo: string,
+    @Ctx() context: Context,
+  ): Promise<User | null> {
+    if (!HasPermissions(context, [PERMISSIONS.USER, PERMISSIONS.ADMIN])) {
+      throw new Error('Error: Unauthorized');
+    }
+
+    if (!phoneNo) {
+      throw new Error('Bad Request: Missing Parameter');
+    }
+
+    try {
+      const user = await UserModel.findOne({phoneNo});
+      if (!user) {
+        throw new Error('Bad Request: User not found');
+      }
+
+      return user;
+    } catch (error) {
+      return error;
```

Comment:
Not a secure practice to directly send the error from server to client without processing. It can in cases reveal unwanted information. Sending a custom string on a basic level is good. Only errors like user not found and others should be sent to the client which the client can understand and also resolve by doing changes on their end, by sending correct user ID. If it is an error like, unhandled promise on server, there isn't anything that client can do, so makes no sense to send it